cmd/snap/cmd_run.go

```diff
@@ -1132,6 +1132,14 @@ func (x *cmdRun) runSnapConfine(info *snap.Info, securityTag, snapApp, hook stri
 		env["XAUTHORITY"] = xauthPath
 	}
 
+	// Guarantee that XDG_RUNTIME_DIR does exist before launching the snap
+	if xdg_runtime_dir, exists := env["XDG_RUNTIME_DIR"]; exists {
+		if err = os.MkdirAll(xdg_runtime_dir, 0700); err != nil {
```

Comment:
I wonder if this should just be `os.Mkdir`.

If the application is not running within a user session, we should not be creating a `/run/user/NNN` directory for it: it'll get mounted over if that user logs in, and an app not running inside a user session should not be accessing XDG_RUNTIME_DIR anyway.

If the real XDG_RUNTIME_DIR exists, we should at most be creating a single directory. Switching to `os.Mkdir` would also require ignoring `os.IsExist(err)`.

Reply:
The mkdir syscall will return EEXIST if you call it with a path that already exists. The `MkdirAll` utility function does not pass on that type of error, so we'd need to explicitly ignore the error when replacing an MkdirAll call with Mkdir.

Reply:
Done.